packages/fuel-gauge/src/contract-factory.test.ts

```diff
@@ -518,4 +518,35 @@ describe('Contract Factory', () => {
       })
     );
   });
+
+  it('deploys large contract via blobs twice and only uploads blobs once', async () => {
+    using launched = await launchTestNode();
+
+    const {
+      wallets: [wallet],
+    } = launched;
+
+    const sendTransactionSpy = vi.spyOn(wallet, 'sendTransaction');
+    const factory = new ContractFactory(LargeContractFactory.bytecode, LargeContract.abi, wallet);
+
+    const firstDeploy = await factory.deployAsBlobTx<LargeContract>({
+      salt: concat(['0x01', new Uint8Array(31)]),
+    });
+    const { contract: firstContract } = await firstDeploy.waitForResult();
+    const firstDeployCalls = sendTransactionSpy.mock.calls.length;
+    const secondDeploy = await factory.deployAsBlobTx<LargeContract>({
+      salt: concat(['0x02', new Uint8Array(31)]),
+    });
+    const { contract: secondContract } = await secondDeploy.waitForResult();
+    const secondDeployCalls = sendTransactionSpy.mock.calls.length;
+    expect(secondDeployCalls - firstDeployCalls).toBeLessThan(firstDeployCalls);
```

Comment:
I can check the exact number of calls if we prefer but potentially prone to flakiness over time. The important thing is that the second deploy submits less txs than the first as the blobs should already be uploaded.

Reply:
Should the fact we are using a node launcher in isolation be a guarantee against flakiness?

Reply:
Flakiness was probably the wrong word here. I mean that if we alter anything in the tx flow so we introduce more or less txs, the test will remain solid as the point is only that the second deploy should use less, rather than checking for a specific number of calls. Right now the number of calls is consistent.